pytools/graph.py

```diff
@@ -520,5 +521,89 @@ def dfs(node: NodeT) -> None:
 
     return visited == graph.keys()
 
+# }}}
+
+
+# {{{ tag propagation graph processing
+
+def get_propagation_graph_from_constraints(
+        constraints: List[Tuple[str, str]],
```

Comment:
```suggestion
        constraints: Sequence[Tuple[str, str]],
```
? Or even just `Iterable`?